### ISSUE TITLE
cannot use custom cellpose model

### ISSUE THREAD

**Issue by pr4deepr** [Vizgen/vizgen-postprocessing#20]
Hi
  Thanks for masking this tool available. 
I would to try using a custom trained cellpose model on a dataset as we are not satisfied with the default segmentation outputs from the instrument after acquisition. I am running this on our HPC. 

I would like to try a custom cellpose model, but I'm running into errors. I used the _cellpose_default_1_ZLevel.json_ file. 
The segmentation works fine if I don't specify custom_weights, but when I specify the path to custom_weight, I get the following error:

```
023-08-18 16:55:32,683 - . - INFO - run run-segmentation with args:Namespace(segmentation_algorithm='segmentation_config_1_z_level.json', input_images='input_images/', input_micron_to_mosaic='input_images/micron_to_mosaic_pixel_transform.csv', output_path='analysis_test_outputs1', tile_size=4096, tile_overlap=None, max_row_group_size=17500, overwrite=False)
2023-08-18 16:55:32,757 - . - INFO - installed versions:
python 3.10.12 | packaged by conda-forge | (main, Jun 23 2023, 22:40:32) [GCC 12.3.0]
vpt 1.1.0
vpt_core 1.0.0
vpt-plugin-cellpose 1.0.0
vpt-plugin-watershed 1.0.0
geopandas 0.12.1
shapely 2.0.0
s3fs 2021.10.0
gcsfs 2021.10.0
rasterio 1.3.0
cellpose 1.0.2
stardist 0.8.3
2023-08-18 16:55:32,764 - . - INFO - run_segmentation started
2023-08-18 16:55:32,922 - . - INFO - prepare segmentation started
2023-08-18 16:55:38,152 - . - INFO - prepare segmentation finished
2023-08-18 16:55:38,157 - . - INFO - Run segmentation on tile 0 started
2023-08-18 16:55:38,158 - . - INFO - Tile 0 [0, 0, 4914, 4914]
2023-08-18 16:55:38,359 - . - ERROR - exception of type <class 'RuntimeError'> thrown: Error(s) in loading state_dict for CPnet:
        Unexpected key(s) in state_dict: "diam_mean", "diam_labels". 
2023-08-18 16:55:38,360 - . - ERROR - vpt has encountered a runtime error (Error(s) in loading state_dict for CPnet:
        Unexpected key(s) in state_dict: "diam_mean", "diam_labels". ) and will now exit
```
**The model was trained in cellpose 2.0. Would this be the cause of the problem?** 

Cheers
Pradeep

**Comment by pr4deepr** [comment]
Used an updated version of cellpose and specified the channel order for segmentation in vpt-plugin-cellpose

**Comment by pakiessling** [comment]
Hi @pr4deepr,

how did you fix this?

vpt-cellpose-plugin seems to require cellpose = "1.0.2"

and what did you change with the channels?

**Comment by pr4deepr** [comment]
I changed the  'pyproject.toml' file and removed the cellpose and torch requirement. Installed cellpose separately and then installed vpt cellpose plugin with the modified requirements.

There were some conflicts with numpy, and other libraries though which I resolved manually. 
This should be enough to get cellpose running. Just verify you can run 
'vpt --help' and 'cellpose' separately without any errors.. Depending on the model you use, you may have to edit the 'predict.py' file to include parameters from [here](https://cellpose.readthedocs.io/en/latest/api.html#cellpose.models.CellposeModel.eval)
